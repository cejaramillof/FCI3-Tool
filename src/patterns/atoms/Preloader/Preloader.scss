.ringlete-loader {
  background-repeat: no-repeat;
  min-height: 64px;
  background-image: url('Ringlete.svg');
  min-width: 64px;

  > * {
    display: none;
  }
}