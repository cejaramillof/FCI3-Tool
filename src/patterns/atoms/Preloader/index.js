import React from 'react';
import './Preloader.scss'

const Preloader = ({ children = null, isLoading = true, type = 'ringlete' }) => {

  const styleTypes = {
    ringlete: () => <div className="ringlete-loader" />
  }

  return (
    <>
      {isLoading ? styleTypes[type]() : children}
    </>
  )
}

export default Preloader;