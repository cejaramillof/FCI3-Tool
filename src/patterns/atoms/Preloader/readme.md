**Atoms**

# Preloader


Use Preloader for Show/Hide loader animation.


```jsx 
import Preloader from 'patterns/atoms/Preloader';

  <Preloader>
    elements you want to hide if is loading
  </Preloader>
```

Uses ```isLoading``` to toggle visibility of animation loader
Params:

| props             | Type        |                                        |          |
| ----------------- |:-----------:| :--------------------------------------| :--------|
| children          | Component   | Component to render if not laoding     | Optional |
| isLoading         | Boolean     | Toggle show/hide preloader animation   | Optional |
| type              | String      | Type of animation                      | Optional |

```jsx 
import Preloader from 'patterns/atoms/Preloader';

  <Preloader isLoading={true}>
    elements are hidden until isLoading is set to true
  </Preloader>
```