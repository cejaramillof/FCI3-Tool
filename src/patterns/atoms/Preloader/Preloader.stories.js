import React from 'react';
import { withKnobs, text, boolean } from "@storybook/addon-knobs";
import centered from '@storybook/addon-centered/react';
import Preloader from './index';
import readme from './readme.md';
import 'index.scss';


export default {
  title: 'Atoms/Preloader ',
  parameters: {
    notes: readme,
  },
  component: Preloader,
  decorators: [withKnobs, centered]
};

export const Basic = () => <Preloader isLoading={boolean('isLoading', true)}>{text("Hidden elements", "Content loaded")}</Preloader>;

