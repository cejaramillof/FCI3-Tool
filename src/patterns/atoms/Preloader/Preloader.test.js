import React from 'react';
import { mount } from 'enzyme';
import Preloader from './index';

let wrapper = mount(
  <Preloader>
    <div id="children">Dolore in esse esse fugiat dolor minim eu reprehenderit excepteur est.</div>
  </Preloader>
);

describe('Preloader atom component', () => {

  it('Should render correctly', () => expect(wrapper).toMatchSnapshot())
  wrapper.setProps({ isLoading: false });
  it('Should render children if is not loading', () => expect(wrapper.find('#children')).toHaveLength(1));
})
