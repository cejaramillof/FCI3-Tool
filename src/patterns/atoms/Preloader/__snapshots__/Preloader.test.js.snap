// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader atom component Should render correctly 1`] = `ReactWrapper {}`;
